config/checkstyle/checkstyle.xml

```diff
@@ -1,22 +1,149 @@
+            <!-- Use io.vavr.control.Option instead. -->
+            <property name="illegalClasses" value="java.util.Optional"/>
+            <!-- Use org.checkerframework.checker.nullness.qual.Nullable instead. -->
+            <property name="illegalClasses" value="javax.annotation.Nonnull, javax.annotation.Nullable"/>
+            <property name="illegalClasses" value="org.jetbrains.annotations.NotNull, org.jetbrains.annotations.Nullable"/>
+        </module>
+        <module name="IllegalThrows"/>
+        <module name="IllegalType"/>
+        <module name="InnerAssignment"/>
+        <module name="LeftCurly">
+            <property name="tokens"
+                      value="ANNOTATION_DEF, CLASS_DEF, CTOR_DEF, ENUM_CONSTANT_DEF, ENUM_DEF,
+                    INTERFACE_DEF, LAMBDA, LITERAL_CASE, LITERAL_CATCH, LITERAL_DEFAULT,
+                    LITERAL_DO, LITERAL_ELSE, LITERAL_FINALLY, LITERAL_FOR, LITERAL_IF,
+                    LITERAL_SWITCH, LITERAL_SYNCHRONIZED, LITERAL_TRY, LITERAL_WHILE, METHOD_DEF,
+                    OBJBLOCK, STATIC_INIT"/>
+        </module>
+        <module name="MissingSwitchDefault"/>
+        <module name="MultipleStringLiterals">
+            <property name="allowedDuplicates" value="3"/>
+            <!-- Allow for more than 3 occurrences of an empty string, a single-space string or ", " string in the given file -->
+            <property name="ignoreStringsRegexp" value="^&quot;(| |, )&quot;$"/>
+        </module>
+        <module name="MultipleVariableDeclarations"/>
+        <module name="ModifiedControlVariable"/>
+        <module name="ModifierOrder"/>
+        <module name="MutableException"/>
+        <module name="NeedBraces">
+            <property name="tokens"
+                      value="LITERAL_DO, LITERAL_ELSE, LITERAL_FOR, LITERAL_IF, LITERAL_WHILE"/>
+        </module>
+        <module name="NestedTryDepth"/>
+        <module name="NoWhitespaceAfter"/>
+        <module name="NoWhitespaceBefore"/>
+        <module name="OneStatementPerLine"/>
+        <module name="PackageDeclaration"/>
+        <module name="ParameterAssignment"/>
+        <module name="RedundantModifier"/>
+        <module name="Regexp">
+            <property name="illegalPattern" value="true"/>
+            <property name="ignoreComments" value="true"/>
+            <!-- Sometimes the assertion expression spans over more than one line, so we must include newlines in the regex. -->
+            <property name="format" value="\bassert\b([^:]|\n)*;"/>
+            <property name="message" value="Assertion without comment/message"/>
+        </module>
+        <module name="Regexp">
+            <property name="illegalPattern" value="true"/>
+            <property name="ignoreComments" value="true"/>
+            <property name="format" value="\bjava\.util\.Date\b"/>
+            <property name="message" value="Usage of the prohibited and deprecated java.util.Date. Please use java.time.Instant instead"/>
```

Comment:
We cannot `archunitize` this one, can we?

Reply:
Whoops right

Reply:
We can